tests/controllers/feature/feature_test.go

```diff
@@ -0,0 +1,187 @@
+package feature_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/feature"
+	"github.com/rancher/rancher/pkg/features"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/wrangler/v2/pkg/crd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+type FeatureTestSuite struct {
+	suite.Suite
+	ctx             context.Context
+	cancel          context.CancelFunc
+	testEnv         *envtest.Environment
+	wranglerContext *wrangler.Context
+}
+
+const (
+	tick     = 1 * time.Second
+	duration = 10 * time.Second
+)
+
+func (s *FeatureTestSuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithCancel(context.TODO())
+	// Load CRD from YAML for REST Client
+	s.testEnv = &envtest.Environment{}
+	restCfg, err := s.testEnv.Start()
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), restCfg)
+
+	// Create CRDs
+	factory, err := crd.NewFactoryFromClient(restCfg)
+	assert.NoError(s.T(), err)
+	err = factory.BatchCreateCRDs(s.ctx,
+		crd.CRD{
+			SchemaObject: v3.Feature{},
+			NonNamespace: true,
+		},
+		crd.CRD{
+			SchemaObject: v3.NodeDriver{},
+			NonNamespace: true,
+		},
+	).BatchWait()
+	assert.NoError(s.T(), err)
+
+	// Create the wrangler context
+	s.wranglerContext, err = wrangler.NewContext(s.ctx, nil, restCfg)
+	assert.NoError(s.T(), err)
+
+	// Register the feature controller
+	feature.Register(s.ctx, s.wranglerContext)
+
+	// Create and start the feature controller factory
+	cf := s.wranglerContext.ControllerFactory.ForResourceKind(schema.GroupVersionResource{
+		Group:    "management.cattle.io",
+		Version:  "v3",
+		Resource: "features",
+	}, "Feature", false)
+	assert.NoError(s.T(), err)
+	err = cf.Start(s.ctx, 1)
+	assert.NoError(s.T(), err)
+
+	// Start the NodeDriver cache
+	_, err = s.wranglerContext.ControllerFactory.SharedCacheFactory().ForKind(schema.GroupVersionKind{
+		Group:   "management.cattle.io",
+		Version: "v3",
+		Kind:    "NodeDriver",
+	})
+	assert.NoError(s.T(), err)
+	err = s.wranglerContext.ControllerFactory.SharedCacheFactory().StartGVK(s.ctx, schema.GroupVersionKind{
+		Group:   "management.cattle.io",
+		Version: "v3",
+		Kind:    "NodeDriver",
+	})
+	assert.NoError(s.T(), err)
+
+	// The feature controller checks the harvester NodeDriver, so it needs to exist
+	n, err := s.wranglerContext.Mgmt.NodeDriver().Create(&v3.NodeDriver{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester",
+		},
+	})
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), n)
+
+	// Initialize Features
```

Comment:
How about...

```go
// Initialize Features - this creates feature resources including the "harvester" feature.
```

It wasn't obvious to me how the feature that was being synced was being created.